dmoj/executors/RUST.py

```diff
@@ -56,15 +57,44 @@ def create_files(self, problem_id, source_code, *args, **kwargs):
         with open(self._file('Cargo.toml'), 'wb') as f:
             f.write(CARGO_TOML)
 
+    shared_target = None
+
+    @classmethod
+    def get_shared_target(cls):
+        if cls.shared_target is not None:
+            return cls.shared_target
+
+        cargo_dir = os.path.expanduser('~/.cargo')
+        collisions = 0
+        while True:
+            maybe_target = cargo_dir + f'/dmoj-shared-target-{collisions}'
```

Comment:
By the way can we use proper path concatenation?

Reply:
Do you mean `os.path.join`, or `pathlib`, or something else?

Reply:
Using `os.path.join`.